opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1000,21 +1001,32 @@ namespace Opm {
                 fip_.fip[i].resize(nc,0.0);
             }
 
-            for (int c = 0; c < nc; ++c) {
-                const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
+            ElementContext elemCtx(ebosSimulator_);
+            auto elemIt = elemCtx.gridView().template begin</*codim=*/0>();
+            const auto& elemEndIt = elemCtx.gridView().template end</*codim=*/0>();
+            for (; elemIt != elemEndIt; ++elemIt) {
```

Comment:
Would not this loop be better as
```
for (const auto& elem : elements(elemCtx.gridView(), Dune::InteriorEntity)) {
```
That would also avoid the if() below.

Also: is this verified to be correct in parallel? (It looks reasonable, but I am not sure.)